xprof/btx_timeline.cpp

```diff
@@ -72,78 +78,122 @@ static perfetto_uuid_t get_parent_counter_track_uuid(timeline_dispatch_t *dispat
+                                               std::unordered_map<hp_ddomain_t, perfetto_uuid_t> &counter_tracks,
+                                               const std::string &track_name, const std::string &hostname, uint64_t process_id,
+                                               thapi_device_id did, uint32_t deviceIdx, uint64_t tHandle, thapi_domain_idx domain,
+                                               std::optional<Details> details = std::nullopt,
+                                               std::unordered_map<hp_dfsdev_t, perfetto_uuid_t> *counter_tracks_fp = nullptr) {
   perfetto_uuid_t hp_dev_uuid = 0;
-  auto [it, inserted] = counter_tracks.insert({{hostname, process_id, did, domain}, hp_dev_uuid});
-  auto &potential_uuid = it->second;
-  // Exists
-  if (!inserted)
-    return potential_uuid;
-
-  perfetto_uuid_t hp_uuid = get_parent_counter_track_uuid(dispatch, hostname, process_id, did);
-  hp_dev_uuid = gen_perfetto_uuid();
-  potential_uuid = hp_dev_uuid;
+  perfetto_uuid_t hp_uuid;
+
+    if (details && counter_tracks_fp != nullptr) {
+    auto [it, inserted] = counter_tracks_fp->insert({{hostname, process_id, did, tHandle, domain,  details->RxTx},
+                                                    hp_dev_uuid});
+    auto &potential_uuid = it->second;
+    if (!inserted)
+      return potential_uuid;
+
+    hp_uuid = get_parent_counter_track_uuid(dispatch, hostname, process_id, did, deviceIdx);
+    hp_dev_uuid = gen_perfetto_uuid();
+    potential_uuid = hp_dev_uuid;
+  } else {
+    auto [it, inserted] = counter_tracks.insert({{hostname, process_id, did, domain, tHandle}, hp_dev_uuid});
+    auto &potential_uuid = it->second;
+    if (!inserted)
+      return potential_uuid;
+
+    hp_uuid = get_parent_counter_track_uuid(dispatch, hostname, process_id, did, deviceIdx);
+    hp_dev_uuid = gen_perfetto_uuid();
+    potential_uuid = hp_dev_uuid;
+  }
 
-  // Create new track
   auto *packet = dispatch->trace.add_packet();
   packet->set_timestamp(0);
   packet->set_trusted_packet_sequence_id(TRUSTED_PACKED_SEQUENCE_ID);
   auto *track_descriptor = packet->mutable_track_descriptor();
   track_descriptor->set_uuid(hp_dev_uuid);
   track_descriptor->set_parent_uuid(hp_uuid);
   std::ostringstream oss;
-  oss << track_name << " | Domain " << domain;
+  if (track_name=="FabricT" && details) {
+    oss << track_name << " | SD " << domain;
+    oss << " | " << details->fabricId << "<->" << details->remotePortId << " | " <<(details->RxTx ? " TX" : " RX");
+  }
+  else if (track_name==" Memory" && details) {
+    oss << track_name << " Module " << domain;
+    oss <<  " | " <<(details->RxTx ? "WR BW" : "RD BW");
+  }
+  else if (track_name=="Allocated Memory (%)") {
+    oss << track_name << " Module " << domain;
+  }
+  else if (track_name==" CopyEngine (%)" || track_name==" ComputeEngine (%)") {
+    oss << track_name << " | SubDevice " << domain;
+  }
+  else if (track_name=="   Power") {
```

Comment:
Why not only one space?

Reply:
If you have more leading space you would be the first. If power takes one space, the others' order will alphabetical and would lose the order. I agree I need more robust way to do it.